starwars/src/App.js

```diff
@@ -22,17 +28,46 @@ class App extends Component {
         return res.json();
       })
       .then(data => {
-        this.setState({ starwarsChars: data.results });
+        this.setState({
+          starwarsChars: data.results,
+          previousPage: data.previous,
+          nextPage: data.next
+        });
       })
       .catch(err => {
         throw new Error(err);
       });
   };
 
+  nextPage = () => {
+    if (this.state.nextPage === null) {
+      this.getCharacters(this.state.currentPage);
+    } else {
+      this.getCharacters(this.state.nextPage);
+    }
+  };
+
+  previousPage = () => {
+    if (this.state.previousPage === null) {
+      this.getCharacters(this.state.currentPage);
+    } else {
+      this.getCharacters(this.state.previousPage);
+    }
+  };
+
   render() {
     return (
       <div className="App">
         <h1 className="Header">React Wars</h1>
+        <div className="button-container">
+          <button className="btn" onClick={this.previousPage}>
```

Comment:
```suggestion
          <button className="btn" onClick={this.previousPage} disabled={this.state.previousPage === null}>
```
You'd have to set a disabled style, but this will keep the button from being pressable at all if this.state.previousPage is null